stackrox/stackrox
# ROX-19081: Add conditional RBAC in Reports 1.0

## Description

### Analysis

1. `NotifierName` and `NotifierSelection` require READ_ACCESS for Integration resource.
2. `VulnMgmtReportDetail` redundantly requires READ_WRITE_ACCESS for WorkflowAdministration
    * It requires READ_WRITE_ACCESS for Access, which is obsolete.
    * It requires READ_WRITE_ACCESS for Integration, even though VulnMgmtReportForm provides conditional prop for notifiers.
    * It does not require READ_ACCESS for Image although VulnMgmtReportTablePage requires it.
3. `ResourceScopeFormModal` is no longer used.

### Solution

1. Add `'Integration'` to `resourceAccessRequirements` in routePaths.ts file.
2. Delete redundant requirements.
3. Call `hasWriteAccessForReport` function as single source of truth.
    * ROX-19085 tracks removal of the incorrect Access resource requirement from run API
4. Delete conditional rendering to edit report from `ScopeName` component.

### Residue

1. Verify whether `AccessControlScope` can be deleted.
2. Fix ambiguity with multiple **Vulnerability Management** expandable items in reportingTable integration test.
3. `SeverityLabelsList` renders color text instead of normal text with color icons, with yet another variation on severity colors.

## Checklist

## Testing Performed

Deletion and refactoring decreased the amount of conditional code in container, but rearrangement caused an increase in main bundle.

1. `yarn lint` in ui
2. `yarn build` in ui and then compute branch - master for the following
    * `wc build/static/js/*.js`
        main.css: 14 = 3759702 - 3759688
        total: -119 = 9948897 - 9949016
    * `ls -al build/static/js/*.js | wc -l`
        0 = 81 - 81 files
3. `yarn start` in ui

### Manual testing

1. See **Vulnerability Management (1.0) Reporting** only if resources for route.

2. Visit /main/vulnerability-management/reports

    * See **Create report** button only if minimum resources to write.

    * Click **Create report** button and see **Create collection** button but **Create email notifier** button only if additional resource.

    * Click **Run report now** and observe that request requires READ_ACCESS for Access resouce.

        ```ts
        // Run report now action requires Access and Image resources.
        // Access seems like a leftover from access scope as report scope.
        ```
